### ISSUE TITLE
Issue with using clause with null-coalescing operator

### ISSUE THREAD

**Issue by jasonwoods-7** [Fody/InfoOf#440]
#### You should already be a Patron

I am a Patron

#### Preamble

Found an issue which throws `System.InvalidProgramException: Common Language Runtime detected an invalid program.`

#### Describe the issue

The following code causes the exception to be thrown when run:

```csharp
using System.IO;

class Foo
{
    Stream nullStream = null;

    void Bar()
    {
        using var _ = nullStream ?? Stream.Null;
        var myInfo = Info.OfMethod<Foo>(nameof(Bar));
    }
}
```

#### Minimal Repro

I've added some failing unit tests to jasonwoods-7/InfoOf@e47aa1e12b6f9cecb247bcf7ed7960a482e1b778

#### Make an effort to fix the bug

I was able to fix 2 of these issues by adding this code jasonwoods-7/InfoOf@ccefba4d66c68cab56c1939e24a535668b746bbd

However, this still results in 12 tests still failing.

I think the issue is caused by a `brtrue.s` instruction pointing to the wrong instruction.

Here is the before\after IL diff of a failing case:

```diff
@@ -1,43 +1,44 @@
 .method /* 060000B9 */ public hidebysig 
     instance class [System.Runtime]System.Reflection.ConstructorInfo GetOfConstructorTypedWithParam_UsingBlock () cil managed 
 {
-    // Method begins at RVA 0x3e14
-    // Code size 43 (0x2b)
-    .maxstack 2
+    // Method begins at RVA 0x3e1c
+    // Code size 53 (0x35)
+    .maxstack 3
     .locals /* 11000007 */ init (
-        [0] class [System.Runtime]System.IO.Stream _,
+        [0] class [System.Runtime]System.IO.Stream,
         [1] class [System.Runtime]System.Reflection.ConstructorInfo
     )
 
     IL_0000: nop
     IL_0001: ldarg.0
-    IL_0002: ldfld class [System.Runtime]System.IO.Stream class WithBlocksGeneric`1<!T>::nullStream /* 0A00002C */
+    IL_0002: ldfld class [System.Runtime]System.IO.Stream class WithBlocksGeneric`1<!T>::nullStream /* 0A000018 */
     IL_0007: dup
-    IL_0008: brtrue.s IL_0010
+    IL_0008: brtrue.s IL_0011
 
     IL_000a: pop
-    IL_000b: ldsfld class [System.Runtime]System.IO.Stream [System.Runtime]System.IO.Stream::Null /* 0A000029 */
-
+    IL_000b: ldsfld class [System.Runtime]System.IO.Stream [System.Runtime]System.IO.Stream::Null /* 0A000015 */
     IL_0010: stloc.0
     .try
     {
-        IL_0011: ldstr "Int32" /* 7000029C */
-        IL_0016: call class [System.Runtime]System.Reflection.ConstructorInfo [InfoOf]Info::OfConstructor<class WithBlocksGeneric`1<int32>>(string) /* 2B000018 */
-        IL_001b: stloc.1
-        IL_001c: leave.s IL_0029
+        IL_0011: ldtoken method instance void WithBlocksGeneric`1::.ctor(int32) /* 060000A5 */
+        IL_0016: ldtoken WithBlocksGeneric`1 /* 0200000B */
+        IL_001b: call class [System.Private.CoreLib]System.Reflection.MethodBase [System.Private.CoreLib]System.Reflection.MethodBase::GetMethodFromHandle(valuetype [System.Private.CoreLib]System.RuntimeMethodHandle, valuetype [System.Private.CoreLib]System.RuntimeTypeHandle) /* 0A000008 */
+        IL_0020: castclass [System.Private.CoreLib]System.Reflection.ConstructorInfo /* 01000016 */
+        IL_0025: stloc.1
+        IL_0026: leave.s IL_0033
     } // end .try
     finally
     {
-        IL_001e: ldloc.0
-        IL_001f: brfalse.s IL_0028
+        IL_0028: ldloc.0
+        IL_0029: brfalse.s IL_0032
 
-        IL_0021: ldloc.0
-        IL_0022: callvirt instance void [System.Runtime]System.IDisposable::Dispose() /* 0A00002A */
-        IL_0027: nop
+        IL_002b: ldloc.0
+        IL_002c: callvirt instance void [System.Runtime]System.IDisposable::Dispose() /* 0A000016 */
+        IL_0031: nop
 
-        IL_0028: endfinally
+        IL_0032: endfinally
     } // end handler
 
-    IL_0029: ldloc.1
-    IL_002a: ret
+    IL_0033: ldloc.1
+    IL_0034: ret
 } // end of method WithBlocksGeneric`1::GetOfConstructorTypedWithParam_UsingBlock
```

The problem is on the `IL_0008` line. It seems this instruction should remain pointing to the `IL_0010` instruction, but it gets updated to the `IL_0011` instruction.

I've traced through the code, but nowhere in the weaver is it updating the operand of that instruction. I'm not sure if this is a Cecil issue, or if there's something else I could be doing to make sure the instruction remains unchanged when Cecil rewrites the assembly.

**Comment by tom-englert** [comment]
Got the root cause. Just need to clean up everything.

**Comment by tom-englert** [comment]
Fixed in 2.1.1
